brave/src/test/java/brave/features/propagation/SecondaryConditionalSamplingTest.java

```diff
@@ -28,13 +29,113 @@
+ *
+ * <p>Since Edgar cannot use the primary sampling state used by B3, it can only use a secondary
+ * (extra field). When Edgar decides to record data, it must set {@link TraceContext#sampledLocal()}
+ * so that this decision can override any unsampled decision in B3, but not actually change B3 state
+ * in the process. Finally, we set {@link Tracing.Builder#alwaysReportSpans()} to ensure data from
+ * Edgar goes to the same span reporter as Salp.
+ *
+ * <p>Here's a simplified example of extra field handling from a sampling point of view:
+ * <pre>{@code
+ * class EdgarSampler extends ExtraFieldPropagation.Customizer {
+ *   @Override
+ *   public FieldCustomizer extractCustomizer(TraceContextOrSamplingFlags.Builder builder) {
+ *     return (fieldName, value) -> {
+ *       // look at the extra field to see if it means we should boost the signal
+ *       if (fieldName.equals("edgar") && edgarWants(value)) builder.sampledLocal();
+ *       return value;
+ *     };
+ *   }
+ * }
+ * }</pre>
+ *
+ * <h2>Reporting data eventually to multiple systems</h2>
+ * In normal Zipkin, sampling is implicit from a backend POV. If you received data, then it must
+ * have been sampled. In code, when {@link TraceContext#sampled()} is true, data reports to the
+ * {@link Tracing.Builder#spanReporter(Reporter) normal span reporter}, and most backends assume the
+ * entire trace will be present.
+ *
+ * <p>Now, we set {@link Tracing.Builder#alwaysReportSpans()} to allow Edgar's conditionally
+ * sampled data reach Mantis. If we do nothing else, we have a concern which is that Edgar data ends
+ * up in Salp's storage even though it is incomplete from a trace POV. This could cause questions
+ * and will in any case add load to Salp's storage.
+ *
+ * <p>To make this concrete, any of the below scenarios will now result in reported spans:
+ * <ul>
+ *   <li>Salp sampled, but Edgar not</li>
+ *   <li>Salp unsampled, but Edgar sampled</li>
+ *   <li>Both sampled</li>
+ * </ul>
+ *
+ * The only trouble area is when Salp says no, but Edgar says yes. If we had more than one secondary
+ * tracing system, differentiation could be a little trick because multiple systems could trigger a
+ * {@link TraceContext#sampledLocal() sampled local decision}. Luckily, here there is no ambiguity:
+ * if {@link TraceContext#sampledLocal()} and not {@link TraceContext#sampled()}, it was Edgar!
+ *
+ * <p>Ex.
+ * <pre>{@code
+ * class AddSampledTag extends FinishedSpanHandler {
+ *   // Propagate a data routing hint to Mantis to ensure Edgar doesn't end up flooding Salp when
+ *   // Salp says no.
+ *   @Override public boolean handle(TraceContext context, MutableSpan span) {
+ *     boolean sampledToSalp = Boolean.TRUE.equals(context.sampled()); // primary remote sampling
+ *     boolean sampledToEdgar = context.sampledLocal(); // secondary overlay
+ *     span.tag("sampled", sampledToSalp && sampledToEdgar ? "salp,edgar"
```

Comment:
here we want to ensure edgar doesn't get irrelvant data, and salp doesn't get partial data from edgar. The interesting case is when both say yes, which means same json is should go to both places.